packages/recommend-react/src/useAlgoliaAgent.ts

```diff
@@ -5,11 +5,15 @@ import { version } from './version';
 
 type UseAlgoliaAgentProps = Pick<
   GetRelatedProductsProps<any>,
-  'recommendClient'
+  'recommendClient' | 'initialState'
 >;
 
 export function useAlgoliaAgent(props: UseAlgoliaAgentProps) {
   useEffect(() => {
-    props.recommendClient.addAlgoliaAgent('recommend-react', version);
-  }, [props.recommendClient]);
+    if (props.initialState) {
+      props.recommendClient.addAlgoliaAgent('recommend-server', version);
```

Comment:
I'm not sure that's a correct heuristic. You could provide `initialState` and not be on the server.

Reply:
@Haroenv @sarahdayan Do you have opinions about this?

Reply:
the server agent indeed should be added on the place that _sends_ the request, not the place that _receives_ the initialState

Reply:
Yes, I think this is too early to add this agent. The PR isn't about SSR, but about initial state. Even if the initial state API is used by SSR, we're mixing concerns here that are unrelated.